dashboard/modules/runtime_env/runtime_env_agent.py

```diff
@@ -243,7 +318,9 @@ def setup_plugins():
                         "Runtime env already failed. "
                         f"Env: {serialized_env}, err: {error_message}"
                     )
-                    return runtime_env_agent_pb2.CreateRuntimeEnvReply(
+                    if request.source_process not in self.reference_exclude_sources:
+                        self.decrease_reference_for_uris(uris)
```

Comment:
The python code with try/except seems to make it difficult to make correct atomic ref counting. Why don't we refactor code in this way?

```python3
increase_ref_count(runtime_env)
status = create_runtime_env(runtime_env)
if status != success:
    decrease_ref_count(runtime_env)

reply(status)
    

```